libs/frontend/ui-build/src/components/StyleRow.vue

```diff
@@ -173,6 +214,42 @@ const removeStyle = () => {
   emit('remove')
 }
 
+const onBackgroundAssetSelected = (asset: ISiteAssetViewModel) => {
+  const assetUrl = urlFromAsset(asset)
+  if (style.value.property === Css.Background) {
+    const newValue = replaceBackground(assetUrl)
+    updateValue(newValue)
+    saveStyle()
+  } else if (style.value.property === Css.BackgroundImage) {
+    updateValue(`url("${assetUrl}")`)
+    saveStyle()
+  }
+  showSelectAssetModal.value = false
+}
+
+const replaceBackground = (assetUrl: string) => {
+  const cssValue = `url("${assetUrl}")`
+  const currentValue = style.value.value
+  if (CSS.supports('background', currentValue)) {
+    // Replace the background value and retain any non-url information.
+    // Find a more efficient & comprehensive way to extract attributes from CSS value.
+    const attributes = style.value.value
+      .split(/([^\s]+|"[^"]+"|'[^']+')/g)
```

Comment:
Hi @sampullman 

I found it difficult to correctly extracts all attributes from a valid `background` css value; as a developer I really hate these shorthand properties 😅. It took me a whole day to finally come up with a somewhat okay regex.

Splitting the value with a space couldn't work because spaces are allowed in some of the [constituent properties](https://developer.mozilla.org/en-US/docs/Web/CSS/background#constituent_properties) of `background`. The following is a list of the example `background` values on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/background):

- `green`
- `content-box radial-gradient(crimson, skyblue)`
  - `url("...")` cannot be used together with `radial-gradient(crimson, skyblue)`, so we have to find where it is and replace it.
- `no-repeat url("../../media/examples/lizard.png")`
- `left 5% / 15% 60% repeat-x url("../../media/examples/star.png")`
- `center / contain no-repeat url("../../media/examples/firefox-logo.svg")`

Reply:
I see! I think it would have been acceptable to only support the `background-image` property for this feature, but this seems to work well.

If there end up being problems we can extract the function and add some unit tests.